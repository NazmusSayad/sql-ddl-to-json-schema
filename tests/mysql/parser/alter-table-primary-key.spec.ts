import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should alter table adding primary key with index options, two columns and options.': {
    queries: [
      `ALTER TABLE people add constraint pk_id__o_id primary key using btree ( id ( 2 ), o_id ( 3 ) asc ) key_block_size 1024 comment 'test';`,
    ],
  },

  'Parser: Should alter table adding primary key with two columns and option.': {
    queries: [
      `ALTER TABLE people add constraint pk_id__o_id primary key(id(2),o_id(3)asc)key_block_size 1024;`,
    ],
  },

  'Parser: Should alter table adding primary key with one column.': {
    queries: [`ALTER TABLE people add constraint pk_id primary key ( id ) ;`],
  },

  'Parser: Should alter table adding unnamed primary key.': {
    queries: [`ALTER TABLE people add primary key(id);`],
  },

  'Parser: Should alter table dropping primary key.': {
    queries: [`ALTER TABLE people drop primary key;`],
  },
});
