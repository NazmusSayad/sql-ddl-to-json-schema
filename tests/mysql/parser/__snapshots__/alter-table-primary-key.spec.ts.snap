// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should alter table adding primary key with index options, two columns and options. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addPrimaryKey",
                    "name": "pk_id__o_id",
                    "index": {
                      "id": "P_INDEX_TYPE",
                      "def": "btree"
                    },
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id",
                          "length": 3,
                          "sort": "asc"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      },
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "comment": "test"
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding primary key with one column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addPrimaryKey",
                    "name": "pk_id",
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding primary key with two columns and option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addPrimaryKey",
                    "name": "pk_id__o_id",
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id",
                          "length": 2
                        }
                      },
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "o_id",
                          "length": 3,
                          "sort": "asc"
                        }
                      }
                    ],
                    "options": [
                      {
                        "id": "O_INDEX_OPTION",
                        "def": {
                          "keyBlockSize": 1024
                        }
                      }
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding unnamed primary key. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addPrimaryKey",
                    "name": null,
                    "index": null,
                    "columns": [
                      {
                        "id": "P_INDEX_COLUMN",
                        "def": {
                          "column": "id"
                        }
                      }
                    ],
                    "options": []
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table dropping primary key. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropPrimaryKey"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;
